peer.go

```diff
@@ -291,8 +291,16 @@ func (p *peer) Start() error {
 
 	// Exchange local and global features, the init message should be
 	// very first between two nodes.
-	if err := p.sendInitMsg(); err != nil {
+	if done, err := p.sendInitMsg(); err != nil {
```

Comment:
In this case, I think we can remove the `sendInitMsg()` function all together. Laying everything out in plane site makes the logic much easier to follow: 
```go
msg := lnwire.InitMsg{...}
doneChan := make(chan struct{}, 1)
p.queueMsg(msg, doneChan)
select {
   case <-doneChan:
   case <-time.After(second*5):
    <error here>
}
 done, nil
```